android/src/main/java/com/reactlibrary/RNPurchasesMath.java

```diff
@@ -0,0 +1,39 @@
+package com.reactlibrary;
+
+// The functionality of this class has been extracted from Java 8 Math class
+class RNPurchasesMath {
+    /**
+     * Returns the sum of its arguments,
+     * throwing an exception if the result overflows an {@code int}.
+     *
+     * @param x the first value
+     * @param y the second value
+     * @return the result
+     * @throws ArithmeticException if the result overflows an int
+     */
+    static int addExact(int x, int y) {
```

Comment:
Are these for ensuring type safety across JS <-> Java?

Reply:
This is just for copying the Java 8 implementation, they use `Math.addExact` and `Math.multiplyExact` so didn't want to modify that. The only thing I have changed from the original Java 8 Period implementation is the Exceptions returned in the parse method, in Java 8 they are `DateTimeException`, but didn't want to create that to not overengineer.

Reply:
:thumbsup: